pkg/util/annotation/annotation.go

```diff
@@ -30,81 +29,71 @@ const (
 )
 
 // HasAnnotationKey returns true if the given object has the given annotation key in its ObjectMeta.
-func HasAnnotationKey(obj metav1.Object, key string) (bool, error) {
+func HasAnnotationKey(obj metav1.Object, key string) bool {
 	if IsNilPointer(obj) {
```

Comment:
Let's remove this `IsNilPointer` check to expose the problem early.

Reply:
Removed

Reply:
Let's try removing `IsNilPointer` check from all the functions in this file. If there's no other references to `IsNilReference`, we should remove its definition as well.